core/provisioning-java/src/test/java/org/apache/syncope/core/provisioning/java/pushpull/LDAPMembershipPullActionsTest.java

```diff
@@ -0,0 +1,116 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.syncope.core.provisioning.java.pushpull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.syncope.common.lib.patch.AnyPatch;
+import org.apache.syncope.common.lib.to.EntityTO;
+import org.apache.syncope.common.lib.to.GroupTO;
+import org.apache.syncope.core.persistence.api.dao.AnyTypeDAO;
+import org.apache.syncope.core.persistence.api.dao.GroupDAO;
+import org.apache.syncope.core.persistence.api.entity.user.UMembership;
+import org.apache.syncope.core.persistence.api.entity.user.User;
+import org.apache.syncope.core.provisioning.api.pushpull.ProvisioningProfile;
+import org.apache.syncope.core.provisioning.java.AbstractTest;
+import org.identityconnectors.framework.common.objects.SyncDelta;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.quartz.JobExecutionException;
+import org.springframework.test.util.ReflectionTestUtils;
+
+public class LDAPMembershipPullActionsTest extends AbstractTest {
+
+    private LDAPMembershipPullActions ldapMembershipPullActions;
+
+    @Mock
+    protected AnyTypeDAO anyTypeDAO;
+
+    @Mock
+    protected GroupDAO groupDAO;
+
+    @Mock
+    private ProvisioningProfile<?, ?> profile;
+
+    @Mock
+    private EntityTO entity;
+
+    @Mock
+    private GroupTO groupEntity;
+
+    @Mock
+    private AnyPatch anyPatch;
+
+    @Mock
+    private SyncDelta syncDelta;
+
+    @Mock
+    private Map<String, Set<String>> membershipsBefore;
+
+    @Mock
+    private UMembership uMembership;
+
+    @Mock
+    private User user;
+
+    @BeforeEach
+    public void before() {
+        ldapMembershipPullActions = new LDAPMembershipPullActions();
+
+        List<UMembership> uMembList = new ArrayList<>();
+        uMembList.add(uMembership);
+
+        Mockito.when(groupDAO.findUMemberships(groupDAO.find(entity.getKey()))).thenReturn(uMembList);
+        Mockito.when(uMembership.getLeftEnd()).thenReturn(user);
+        Mockito.when(user.getKey()).thenReturn("userTestKey");
+
+        ReflectionTestUtils.setField(ldapMembershipPullActions, "groupDAO", groupDAO);
+        ReflectionTestUtils.setField(ldapMembershipPullActions, "membershipsBefore", membershipsBefore);
+    }
+
+    @Test
+    public void testBeforeUpdateWithGroupTOAndEmptyMemberships() throws JobExecutionException {
+        Set<String> memb = new HashSet<>();
+
+        Mockito.when(membershipsBefore.get(uMembership.getLeftEnd().getKey())).thenReturn(memb);
+        ldapMembershipPullActions.beforeUpdate(profile, syncDelta, groupEntity, anyPatch);
+
+        Assertions.assertTrue(groupEntity instanceof GroupTO);
```

Comment:
Same about static import for `Assertions.assertTrue` - see https://github.com/apache/syncope/blob/2_1_X/core/provisioning-java/src/test/java/org/apache/syncope/core/provisioning/java/DerAttrTest.java#L21-L24

Reply:
Fixed